include/faunus/energy.h

```diff
@@ -151,8 +154,408 @@ namespace Faunus
                 return string();
             return textio::header("Energy: " + name) + _info();
         }
+
+        virtual double systemEnergy( const Tpvec &p )
+        {
+            double u = external(p);
+            for ( auto g : spc->groupList())
+                u += g_external(p, *g) + g_internal(p, *g);
+            for ( unsigned int i = 0; i < spc->groupList().size() - 1; i++ )
+                for ( unsigned int j = i + 1; j < spc->groupList().size(); j++ )
+                    u += g2g(p, *spc->groupList()[i], *spc->groupList()[j]);
+            return u;
+        }
+
+        virtual double g2All(const Tpvec & p, Group &g)
+        {
+            double unew = 0.0;
+            for ( auto o : spc->groupList()) {
+                if ( o != &g ) {
+                    unew += g2g(p, *o, g);
+                    if ( unew == pc::infty )
+                        return pc::infty;   // early rejection
+                }
+            }
+            return unew;
+        }
     };
 
+
+class EnergyMatrix {
```

Comment:
Have this in a separate header? If you plan to make use the same code here and in other programs that would make it easier to update. Documentation would be nice.